internal/renderer/renderer_test.go

```diff
@@ -0,0 +1,101 @@
+package renderer
+
+import (
+	"testing"
+
+	"github.com/docker/app/internal/renderer/driver"
+	"github.com/pkg/errors"
+	"gotest.tools/assert"
+	is "gotest.tools/assert/cmp"
+)
+
+type fakeDriver struct{}
+
+func (d *fakeDriver) Apply(s string, settings map[string]interface{}) (string, error) {
+	return s + "fake", nil
+}
+
+type fakeErrorDriver struct{}
+
+func (d *fakeErrorDriver) Apply(s string, settings map[string]interface{}) (string, error) {
+	return s, errors.New("error in driver")
+}
+
+func TestRegisterNilPanics(t *testing.T) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("The code did not panic")
+		}
+		drivers = map[string]driver.Driver{}
+	}()
+	Register("foo", nil)
+}
+
+func TestRegisterDuplicatePanics(t *testing.T) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("The code did not panic")
+		}
+		drivers = map[string]driver.Driver{}
+	}()
+	Register("bar", &fakeDriver{})
+	Register("bar", &fakeDriver{})
+}
+
+func TestRegister(t *testing.T) {
+	fd := &fakeDriver{}
+	Register("baz", fd)
+	defer func() {
+		drivers = map[string]driver.Driver{}
+	}()
```

Comment:
Shouldn't we use some kind of tests tear down function for this?

Reply:
:thinking: just for the one-line ? :thinking:

Reply:
It gets repeated across all tests and could easily be forgotten and mess the subsequent tests though…
Not as easy as `SetUp` and `TearDown` functions provided by some test frameworks but we could for instance use https://golang.org/pkg/testing/#hdr-Main